text/0000-move-pointer.md

```diff
@@ -0,0 +1,103 @@
+- Feature Name: `move_pointer`
+- Start Date: 2016-05-12
+- RFC PR: (leave this empty)
+- Rust Issue: (leave this empty)
+
+# Summary
+[summary]: #summary
+
+Introduce a new pointer type `&move` that logically owns the pointed-to data, but does not control the backing memory. Also introduces the DerefMove trait to provide access to such a pointer.
+
+# Motivation
+[motivation]: #motivation
+
+This provides an elegant solution to passing DSTs by value, allowing `Box<FnOnce>` to work. It also will allow other usecases where trait methods should take self by "value".
+
+# Detailed design
+[design]: #detailed-design
+
+- Add a new pointer type `&move T`
+ - This pointer will get the same lifetime rules as `&` and `&mut` (such that it cannot outlive the allocation)
+ - but, it will allow the value to become invalid before the allocation does.
+ - Deref coercions as per RFC #241 apply.
+ - Variables of type `&move T` will only allow mutating `T` if they themselves are `mut`
+- Add a new unary operation `&move <value>`. With a special case such that `&move |x| x` parses as a move closure, requiring parentheses to parse as an owned pointer to a closure.
+ - This precedence can be implemented by ignoring the `move` when parsing unary operators if it is followed by a `|` or `||` token.
+- Add a new operator trait `DerefMove` to allow smart pointer types to return owned pointers to their contained data. 
+ - A type that implements `DerefMove` cannot implement `Drop` (as `DerefMove` provides equivalent functinality)
+
+```rust
+trait DerefMove: DerefMut
+{
+    /// Return an owned pointer to inner data
+    fn deref_move(&mut self) -> &move Self::Target;
```

Comment:
Good point. I suppose making it unsafe is the cop-out option (which semi-fits, because you'd need unsafe code to implement it).

It can't really be &own, because that "moves" the container into the `deref_move` method, leading to it being dropped before the pointer you return.